src/modules/browse.js

```diff
@@ -97,6 +97,105 @@ function createBrowseUrl(filterName, filterValue, parameters, userParameters, op
   return `${serviceUrl}/browse/${encodeURIComponent(filterName)}/${encodeURIComponent(filterValue)}?${queryString}`;
 }
 
+// Create query params from parameters and options
+function createQueryParams(parameters, options) {
```

Comment:
A lot of code is repeating between this and  `createBrowseUrl` you can split the repeating part into a new function and call that from these two. There is an example in my PR.